### DISCUSSION TITLE
What are the differences between putListingItem and feeds JSON_LISTINGS_FEED

### DISCUSSION THREAD

**Discussion by Naitchi** [amzn/selling-partner-api-models#3989] [General]
I'm trying to automate listing for different sites and I'm having trouble with amazon.

I've done all the 6 requests to create a listing with feeds and JSON_LISTINGS_FEED and when looking for response on how to create the json for my book, I keep finding answer for putListingItem.  (only my json is not working everything else is good)

what are the differences between putListingItem and feeds JSON_LISTINGS_FEED ? Is it worth it for me to switch from feeds to putListingItem ?

My goal is to CRUD my little inventory (~200 items) on amazon

**Comment by chapmanjw** [comment]
They ultimately do the same thing and share the same data model. The JSON_LISTINGS_FEED effectively takes your feed submission and calls the Listings Items API on your behalf. When deciding between the two, it comes down to which access pattern works best for your application. The feed allows you to offload the batch of submissions for asynchronous calls to the Listings Items API with a report of any errors the API returns. Directly calling the Listings Items API allows you to receive those validation errors synchronously. 

With either, you would still leverage the Listings Items API to read back any errors that occur after accepting the submission (such as catalog matching errors and image download errors). And you can leverage notifications to tell you when issues have been added or cleared on your listings.

**Reply by chapmanjw** [reply]
With 200 items, your interactions will probably be simpler using the Listings Items API directly. (This way you don't need to package your updates into a feed submission and then unpack error responses from a feed report).
